electroncash/interface.py

```diff
@@ -150,15 +135,11 @@ def _get_socket_and_verify_ca_cert(self, *, suppress_errors) -> Tuple[Optional[s
         if s is not None:
             try:
                 context = self.get_ssl_context(cert_reqs=ssl.CERT_REQUIRED, ca_certs=ca_path)
-                s = context.wrap_socket(s, do_handshake_on_connect=True)
-                # validate cert
-                if s and self.check_host_name(s.getpeercert(), self.host):
-                    self.print_error("SSL certificate signed by CA")
-                    # it's good, return the wrapped socket
-                    return s, False
-                # bad cert or other shenanigans, return None but inform caller
-                # to try alternate "pinned self-signed cert" code path
-                return None, False
+                context.check_hostname = True
+                s = context.wrap_socket(s, do_handshake_on_connect=True, server_hostname=self.host)
+                self.print_error("SSL certificate signed by CA")
+                # it's good, return the wrapped socket
+                return s, False
```

Comment:
I can't figure out from the official documentation what `SSLContext.wrap_socket` returns in case of a failure. From a quick glance at the source code it seems correct that `context.wrap_socket` will always return a valid socket or error, so this should be OK.
The `if s and ... :` probably never made sense to begin with, or else older versions python had `wrap_socket` return `None,` `False` or a socket with a` __bool__ `method?

Reply:
Yea, I also think it always either returns a socket or raises. My tests so far also confirm this. I can't find a case where `wrap_socket` returns `None`, even when setting `check_hostname` to `False`.